modules/setting/config.go

```diff
@@ -15,8 +15,45 @@ type PictureStruct struct {
 	EnableFederatedAvatar *config.Value[bool]
 }
 
+type OpenWithEditorApp struct {
```

Comment:
Is there perhaps any way to set a custom icon, i.e. as a file path?
Once we have this option, this feature is truly complete.

Reply:
I have left a TODO in code:

```
iconHTML = svg.RenderHTML("gitea-git", 16, "gt-mr-3") // TODO: it could support user's customized icon in the future
```

To support "custom icon", it needs far more changes: it needs an extra field, and needs a new UI editor for the config items (at least, a YAML or JSON editor). It still needs the end user to upload the icon to "somewhere", I guess most users would be quite lazy to do so .....

So at the moment, the default "git" icons looks not bad.

Reply:
SVG icons could be entered as text in that box, e.g. `name=url,icon=<svg/>`.

Reply:
Yup, that's why I think "and needs a new UI editor for the config items (at least, a YAML or JSON editor)".

Let's collect more user feedbacks, to see how to move on.

Reply:
I guess we better already change the syntax to yaml/json now.

Reply:
That's quite easy because it is stored as JSON in database. So, it could be rendered as JSON or YAML in the editor easily. It won't block anything.